### ISSUE TITLE
ST3 support

### ISSUE THREAD

**Issue by mu3** [praveenvijayan/Sublime-Text-2-BIDI#12]

**Comment by amurad** [comment]
Yes, I will really appreciate conversion to ST3 package. There is no other package available for RTL support.

**Comment by ttasovac** [comment]
this would be very helpful indeed

**Comment by shahverd** [comment]
We all need it...

**Comment by BilalBoulifa** [comment]
yes, it would be very helpful

**Comment by BilalBoulifa** [comment]
is there any evolution here ? is the plugin is abandoned ?

**Comment by imdark** [comment]
I made a fix for the package imports for sublime 3 and it works well my pc.
I also made a pull request but since it seems the creator of this plugins abandoned it, you can just move to my fork:
https://github.com/imdark/Sublime-Text-2-BIDI

**Comment by BilalBoulifa** [comment]
thank you imdark, work perfectly on ST3

**Comment by ksonbol** [comment]
@imdark Your fork works with ST3, however it makes text unreadable in other text editor. Letters appear flipped horizontally and words are reversed.

**Comment by BilalBoulifa** [comment]
you can restore the file, reopen the contextual menu in ST and reclick on bidirectional text

**Comment by ksonbol** [comment]
Clicking bidirectional text again fixes the words in other editors but breaks them in ST3. Characters appear flipped horizontally and words reversed but now in ST3. I undid all the changes but problem isn't solved yet.

**Comment by BilalBoulifa** [comment]
yeap it resolve the problem just on ST, because the solution at the begining it is not really a good solution, it is based on fliping the characters in order to link them  and reverse words in order to obtain the correct order of the sentence, the reality is that the solution it's not about adding the support of arabic or other languages but just to simulate, the deal was to print it correctly but not to realy support it.
I am not sure but I think it is not possible to support this function by a plugin, I think it must be implemented natively in ST and I talk about realy supporting the function not just print it correctly
.

**Comment by imdark** [comment]
I have been thinking about it for some time now, it does not seem that sublime text will support this any time soon.
I have some ideas for implementing better support via the plugin by keeping track of previously fixed bidi text and calling the plugin code "on text changed", I have written them in https://github.com/praveenvijayan/Sublime-Text-2-BIDI/issues/10. the main thing that is bothering me is edge cases, and performance.
I don't have much experience in python so help or code review would be welcome.

**Comment by ctf0** [comment]
@imdark ,as an away around how about you add a function which when you press <code>ctrl+c</code> to copy the text ,the plugin will also reverse it back to original ,this way when you paste it somewhere else it wont look reversed ,ofcourse this will also require that the plugin remembers what was the text it changed on each session so it doesn't reverse everything you copy.

**Comment by dotanamor** [comment]
me to worked greate
in he
